File: app-core/src/main/java/code/sdk/core/util/IOUtil.java
package code.sdk.core.util;

import android.content.Context;
import android.content.res.AssetManager;
import android.content.res.Resources;

import java.io.ByteArrayOutputStream;
import java.io.Closeable;
import java.io.InputStream;
import java.io.InputStreamReader;
import java.io.StringWriter;
import java.nio.charset.StandardCharsets;

public class IOUtil {
    public static final int BUFFER_SIZE = 4 * 1024;

    public static String readAssetContent(Context context, String file) {
        try {
            AssetManager assets = context.getAssets();
            InputStream inputStream = assets.open(file);
            return readInputStream(inputStream);
        } catch (Exception e) {
            //ObfuscationStub0.inject();
        }
        return null;
    }

    public static String readRawContent(Context context, int resId) {
        Resources resource = context.getResources();
        InputStream inputStream = resource.openRawResource(resId);
        return readInputStream(inputStream);
    }

    public static String readInputStream(InputStream inputStream) {
        InputStreamReader streamReader = null;
        try {
            streamReader = new InputStreamReader(inputStream, StandardCharsets.UTF_8);
            final StringWriter writer = new StringWriter();
            final char[] buffer = new char[BUFFER_SIZE];
            int charRead = streamReader.read(buffer);
            while (charRead > 0) {
                writer.write(buffer, 0, charRead);
                charRead = streamReader.read(buffer);
            }
            return writer.toString();
        } catch (Exception e) {
            //ObfuscationStub1.inject();
        } finally {
            IOUtil.close(inputStream);
            IOUtil.close(streamReader);
        }
        return null;
    }

    public static byte[] toByteArray(InputStream input) {
        byte[] buffer = null;
        try {
            ByteArrayOutputStream output = new ByteArrayOutputStream();
            buffer = new byte[1024 * 4];
            int n = 0;
            while ((n = input.read(buffer)) != -1) {
                output.write(buffer, 0, n);
            }
            buffer = output.toByteArray();
        } catch (Exception e) {
            e.printStackTrace();
        }
        return buffer;
    }

    public static void close(Closeable closeable) {
        if (closeable != null) {
            try {
                closeable.close();
            } catch (Exception e) {
                //ObfuscationStub2.inject();
            }
        }
    }
}

File: lib-util/src/main/java/code/util/IOUtil.kt
package code.util

import android.content.Context
import java.io.ByteArrayOutputStream
import java.io.Closeable
import java.io.InputStream
import java.io.InputStreamReader
import java.io.StringWriter
import java.nio.charset.StandardCharsets

object IOUtil {
    val BUFFER_SIZE = 4 * 1024
    
    fun readRawContent(context: Context, resId: Int): String? {
        val resource = context.resources
        val inputStream = resource.openRawResource(resId)
        return readInputStream(inputStream)
    }

    private fun readInputStream(inputStream: InputStream): String? {
        var streamReader: InputStreamReader? = null
        try {
            streamReader = InputStreamReader(inputStream, StandardCharsets.UTF_8)
            val writer = StringWriter()
            val buffer = CharArray(BUFFER_SIZE)
            var charRead = streamReader.read(buffer)
            while (charRead > 0) {
                writer.write(buffer, 0, charRead)
                charRead = streamReader.read(buffer)
            }
            return writer.toString()
        } catch (e: Exception) {
            //ObfuscationStub1.inject();
        } finally {
            close(inputStream)
            close(streamReader)
        }
        return null
    }

    
    fun toByteArray(input: InputStream): ByteArray? {
        var buffer: ByteArray? = null
        try {
            val output = ByteArrayOutputStream()
            buffer = ByteArray(1024 * 4)
            var n: Int
            while (input.read(buffer).also { n = it } != -1) {
                output.write(buffer, 0, n)
            }
            buffer = output.toByteArray()
        } catch (e: Exception) {
            e.printStackTrace()
        }
        return buffer
    }

    
    fun close(closeable: Closeable?) {
        if (closeable != null) {
            try {
                closeable.close()
            } catch (e: Exception) {
                //ObfuscationStub2.inject();
            }
        }
    }
}